pubtools/exodus/_hooks/pulp.py

```diff
@@ -0,0 +1,210 @@
+                body = response.json()
+            except:
+                raise outer
+
+            LOG.error("Unsuccessful response from exodus-gw: %s", body)
+            raise
+
+    def _do_request(self, **kwargs):
+        if not self._session:
+            self._session = self._new_session()
+
+        resp = self._session.request(**kwargs)
+        self._unpack_response(resp)
+        return resp
+
+    def _check_cert(self):
+        """Issue request to exodus-gw to identify permissions."""
+
+        if not self._gw_crt:
+            LOG.debug("EXODUS_GW_CERT not found, authentication may fail")
+
+        if not self._gw_key:
+            LOG.debug("EXODUS_GW_KEY not found, authentication may fail")
+
+        auth_url = urljoin(self._gw_url, "/whoami")
+        resp = self._do_request(method="GET", url=auth_url)
+        context = resp.json()
+
+        for user_type, ident in (
+            ("client", "serviceAccountId"),
+            ("user", "internalUsername"),
+        ):
+            typed_ctx = context[user_type]
+            if typed_ctx["authenticated"]:
+                LOG.debug(
+                    "Authenticated with exodus-gw at %s as %s %s (roles: %s)",
+                    self._gw_url,
+                    user_type,
+                    typed_ctx[ident],
+                    [str(role) for role in typed_ctx["roles"]],
+                )
+                break
+        else:
+            LOG.debug("Not authenticated with exodus-gw.")
+
+    def _poll_commit_completion(self, commit, tries=4):
+        """Issues request(s) to exodus-gw to get the task state, returning once
+        it either is FAILED or COMPLETED."""
+        task_url = urljoin(self._gw_url, commit["links"]["self"])
+
+        for i in range(tries):
+            resp = self._do_request(method="GET", url=task_url)
+            task = resp.json()
+
+            if task["state"] not in ("FAILED", "COMPLETE"):
+                if i < tries - 1:
+                    wait_time = (i + 1) * 30 if wait_time < 120 else wait_time
+                    LOG.info(
+                        "Commit %s to exodus-gw at %s in progress. "
+                        "Will repoll in %d seconds [try %s/%s]",
+                        task["id"],
+                        self._gw_url,
+                        wait_time,
+                        i + 1,
+                        tries,
+                    )
+                    time.sleep(wait_time)
+                    continue
+
+                raise RuntimeError(
+                    "Commit %s to exodus-gw at %s timed out"
+                    % (task["id"], self._gw_url)
+                )
+            if task["state"] == "FAILED":
+                raise RuntimeError(
+                    "Commit %s to exodus-gw at %s failed"
+                    % (task["id"], self._gw_url)
+                )
+            if i != 0:
+                LOG.info(
+                    "Exodus config deployment succeeded [try: %s/%s]",
```

Comment:
It's not config deployment.

Also, can we please drop the 'try' part?  I think that makes it look like the task had to be retried several times which means something bad is happening.  In fact, all it really means is that we had to do several HTTP requests until we saw the task complete, which is totally normal.